modules/base/acl/module.py

```diff
@@ -25,494 +23,606 @@ def __init__(self, bot):
+        mappings = ACLevelMappping.get_all(ctx.guild.id)
 
-    @commands.check(check.acl)
-    @acl_group.command(name="get")
-    async def acl_group_get(self, ctx, name: str):
-        """Get ACL group."""
-        group = ACL_group.get(ctx.guild.id, name)
-        if group is None:
-            await ctx.reply(_(ctx, "No such group exists."))
+        if not mappings:
+            await ctx.reply(_(ctx, "No mappings have been set."))
             return
 
-        await ctx.reply(embed=self.get_group_embed(ctx, group))
-
-    @commands.check(check.acl)
-    @acl_group.command(name="add")
-    async def acl_group_add(self, ctx, name: str, parent: str, role_id: int):
-        """Add ACL group.
+        mappings = sorted(mappings, key=lambda m: m.level.name)[::-1]
+        items = [Item(mapping) for mapping in mappings]
 
-        :param name: string matching ``[a-zA-Z-]+``.
-        :param parent: ACL parent group name.
-        :param role_id: Discord role ID.
+        table: List[str] = utils.text.create_table(
+            items,
+            header={
+                "role": _(ctx, "Role"),
+                "level": _(ctx, "Level"),
+            },
+        )
 
-        To unlink the group from the parent, set it to "".
+        for page in table:
+            await ctx.send("```" + page + "```")
 
-        To set up virtual group with no link to Discord roles, set ``role_id``
-        to ``0``.
-        """
-        RE_NAME = r"[a-zA-Z-]+"
-        if re.fullmatch(RE_NAME, name) is None:
+    @check.acl2(check.ACLevel.GUILD_OWNER)
+    @acl_mapping_.command(name="add")
+    async def acl_mapping_add(self, ctx, role: nextcord.Role, level: str):
+        """Add ACL level to role mappings."""
+        try:
+            level: ACLevel = ACLevel[level]
+        except KeyError:
             await ctx.reply(
-                _(ctx, "Bad group name (regex `{regex}`)").format(regex=RE_NAME)
+                _(ctx, "Invalid level. Possible options are: {keys}.").format(
+                    keys=", ".join(f"'{key}'" for key in ACLevel.__members__.keys())
+                )
             )
             return
 
-        if len(parent) == 0:
-            parent = None
+        m = ACLevelMappping.add(ctx.guild.id, role.id, level)
+        if m is None:
+            await ctx.reply(_(ctx, "That role is already mapped to some level."))
+            return
 
-        group = ACL_group.add(ctx.guild.id, name, parent, role_id)
-        await ctx.reply(embed=self.get_group_embed(ctx, group))
-        await guild_log.warning(
+        await ctx.reply(
+            _(ctx, "Role **{role}** will be mapped to **{level}**.").format(
+                role=role.name, level=level.name
+            )
+        )
+        await guild_log.info(
             ctx.author,
             ctx.channel,
-            f"New ACL group '{name}' (parent '{parent}', role ID '{role_id}').",
+            f"New ACLevel mapping from role '{role.name}' to level '{level.name}'.",
```

Comment:
I think this is less confusing for pepople not knowing this function.

"New ACLevel mapping for role '{role.name}' set to level '{level.name}'."